### ISSUE TITLE
How to append source list dynamically?

### ISSUE THREAD

**Issue by mikeevstropov** [clappr/clappr-level-selector-plugin#24]
I have js array of src's. Thank you!

**Comment by me-vlad** [comment]
ref: https://github.com/clappr/clappr-level-selector-plugin/issues/23

**Comment by mikeevstropov** [comment]
Now i'm using this way:
``` javascript
var player = new Clappr.Player({
	parentId	: '#player',
	autoplay	: false,
	sources 	: [
		'http://example.com/source1/index.m3u8',
		'http://example.com/source2/index.m3u8',
		'http://example.com/source3/index.m3u8'
	],
	plugins		: { 'core': [LevelSelector] },
	levelSelectorConfig: {
  		labels: {
			3 : '1080p',
  	    	2 : '720p',
  	    	1 : '480p',
  	    	0 : '360p'
  		}
  	},
});
```
But in bitrate list i see only first src.

**Comment by mikeevstropov** [comment]
Maybe i should create the Blob object with source layers like master.m3u8?
But i think it's wrong way...
I have array on client and i can't get master.m3u8 with layers from server.

**Comment by gabrielirdeb** [comment]
I have the same issue here, anyone ?

**Comment by leandromoreira** [comment]
@mikeevstropov and @gabrielirdeb I don't this is a "real" issue, at the best we can see this as a new feature. In theory your variant (master) playlist should be generated in your server, this would provide a smooth playback change.

Besides, this also raises a lot of questions about the expected behavior:

```javascript
    sources     : [
        'http://example.com/source1/index.m3u8',
        'http://example.com/source2/index.m3u8',
        'http://example.com/source3/index.m3u8'
    ],
```

How should we handle this `sources`?

* play one after the other (binge watch) ? 
* use it as fallback (`a la` html5 video) ? 
* use it to compose a playback with different bitrates ?

Maybe we could force a list of qualities (this might be very easy to do), what do you think @clappr/core ?

```javascript
    levelSelectorConfig: {
        levels: {
            [label: '1080p', src: ''http://example.com/source1/index.m3u8''],
            [label: '1080p', src: ''http://example.com/source1/index.m3u8''],
            [label: '1080p', src: ''http://example.com/source1/index.m3u8''],
            [label: '1080p', src: ''http://example.com/source1/index.m3u8'']
        }
    },
```javascript

**Comment by me-vlad** [comment]
@leandromoreira , @clappr/core
+1 for quality list option 
May be it helps to define avaliable levels for native HLS?
https://github.com/clappr/clappr-level-selector-plugin/issues/23

**Comment by leandromoreira** [comment]
that's right @me-vlad what do you think @cowai ?

**Comment by towerz** [comment]
I agree with @leandromoreira, this would fit better as a specific level selector option. Since the `sources` parameter is handled by a separate plugin, this would cause a number of weird conflicts that we could avoid by handling this separately.

Taking the suggested config, the plugin could switch sources using `load()` by itself, and providing an initial seek value, thus avoiding conflicts with other plugins.

**Comment by cowai** [comment]
@leandromoreira This seems like a good solution. By defining it in the plugin we can write our own hls parsers to pass to the plugin. Nice and clean.

**Comment by leandromoreira** [comment]
I made a #30 for it after that we also need to provide the documentation, this also relies at clappr PR (which I'm doing now :P )
```javascript
    levelSelectorConfig: {
        levels: [
            {id: 0, label: '1080p', src: "http://example.com/source1/index.m3u8"},
            {id: 1, label: '1080p', src: "http://example.com/source1/index.m3u8"},
            {id: 2, label: '1080p', src: "http://example.com/source1/index.m3u8"},
            {id: 3, label: '1080p', src: "http://example.com/source1/index.m3u8"}
        ]
    }
```

**Comment by leandromoreira** [comment]
It is not so simple as I thought

**Comment by cowai** [comment]
@leandromoreira Any news on this?

**Comment by leandromoreira** [comment]
@cowai I tried once but then I saw it's not easy...  ;/ I'll try that later, the problems rises while you trying to change container (and it gets worst on mobile)

**Comment by gabrielirdeb** [comment]
Any news on this?

**Comment by leandromoreira** [comment]
We didn't have time yet to see =/ I'm sorry
